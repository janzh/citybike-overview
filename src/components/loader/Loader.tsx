import React from "react";
import styled from "styled-components";

const texts = {
    loading: "Vent, siden laster",
};

const LoaderSpinner = styled.div`
    border: 16px solid #f3f3f3; /* Light grey */
    border-top: 16px solid #3498db; /* Blue */
    border-radius: 50%;
    width: 120px;
    height: 120px;
    animation: spin 2s linear infinite;

    @keyframes spin {
        0% {
            transform: rotate(0deg);
        }
        100% {
            transform: rotate(360deg);
        }
    }
`;

const LoaderContainer = styled.div`
    display: flex;
    flex-direction: column;
    align-items: center;
`;

const Loader = () => (
    <LoaderContainer>
        <LoaderSpinner />
        <div>
            <p>{texts.loading}</p>
        </div>
    </LoaderContainer>
);

export default Loader;
