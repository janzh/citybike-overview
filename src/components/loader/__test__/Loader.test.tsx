import { render } from "@testing-library/react";
import React from "react";

import Loader from "../Loader";

describe("Loader", () => {
    it("should render without failure", () => {
        const { asFragment } = render(<Loader />);
        expect(asFragment()).toMatchSnapshot();
    });
});
