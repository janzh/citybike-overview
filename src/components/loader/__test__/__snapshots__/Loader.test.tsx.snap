// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader should render without failure 1`] = `
<DocumentFragment>
  <div
    class="sc-guDMob jOKWbK"
  >
    <div
      class="sc-beyTiQ jOPfIP"
    />
    <div>
      <p>
        Vent, siden laster
      </p>
    </div>
  </div>
</DocumentFragment>
`;
